libsolutil/CommonIO.cpp

```diff
@@ -137,7 +137,11 @@ string solidity::util::absolutePath(string const& _path, string const& _referenc
 	if (p.begin() == p.end() || (*p.begin() != "." && *p.begin() != ".."))
 		return _path;
 	boost::filesystem::path result(_reference);
-	result.remove_filename();
+
+	// If filename is "/", then remove_filename() throws.
+	// See: https://github.com/boostorg/filesystem/issues/176
+	if (result.filename() != boost::filesystem::path("/"))
+		result.remove_filename();
```

Comment:
Have you checked how it works for paths like `a/b/c.sol//`?

Reply:
For this one, `remove_filename` essentially removes the trailing `/`, making the resulting path `a/b/c.sol`

The problem only seems to manifest with paths such as `////` , `/////` i.e., at least 4 and only `/`